hack/third-party/capx/go.mod

```diff
@@ -3,11 +3,11 @@
 
 module github.com/nutanix-cloud-native/cluster-api-runtime-extensions-nutanix/external/capx
 
-go 1.23.0
+go 1.24.0
 
-toolchain go1.24.2
+toolchain go1.24.3
 
-require github.com/nutanix-cloud-native/cluster-api-provider-nutanix v1.7.0-beta.2
+require github.com/nutanix-cloud-native/cluster-api-provider-nutanix v1.7.0-beta.4
```

Comment:
nit: it may be clear to make this change + all the other unit test and kustomize changes in a separate PR

Reply:
Please review: https://github.com/nutanix-cloud-native/cluster-api-runtime-extensions-nutanix/pull/1233

Reply:
Please review: https://github.com/nutanix-cloud-native/cluster-api-runtime-extensions-nutanix/pull/1236 for the template related change.